### PR TITLE
Reformulate section about netavark-firewalld-reload.service

### PR DESCRIPTION
**Pull request by dcermak**
### PR creator: Description

The section about the new netavark features was not entirely technically correct, hence the following suggestion to improve it.

cc @danishprakash

### CODE REVIEW

File: container-guide/concepts/containers-podman.xml [hunk 1]
```diff
@@ -403,14 +403,28 @@
 <screen>&prompt.user;podman run --rm -ti suse/sle15</screen>
 <note>
 <title>
-Solving network connection issues with rootful containers
+Solving loosing network access to rootful containers
```
Comment (dmpop) [inline container-guide/concepts/containers-podman.xml:406]:
```suggestion
Fixing the lost network access to rootful containers
```

File: container-guide/concepts/containers-podman.xml [hunk 2]
```diff
@@ -403,14 +403,28 @@
...
 </title>
 <para>
-Rootful containers may occasionally lose network connectivity due to another process deleting the firewall rules configured by Netavark. Starting with version 1.9.0, Netavark comes with the <systemitem>netavark-firewalld-reload.service</systemitem> service that solves the problem. Enable and start the service as follows:
+Rootful containers may occasionally become no longer be accessible from the
+outside. This is caused by firewalld reloading its permanent rules and
+discarding any temporary rules created by podman's networking subsystem (either
+CNI or netavark).
```
Comment (dmpop) [inline container-guide/concepts/containers-podman.xml:412]:
```suggestion
Rootful containers may occasionally become inaccessible from the
outside. The issue is caused by firewalld reloading its permanent rules and
discarding any temporary rules created by &podman;'s networking subsystem (either
CNI or &netavark;).
```

File: container-guide/concepts/containers-podman.xml [hunk 3]
```diff
@@ -403,14 +403,28 @@
...
+
+A one-time workaround is to reload the &podman; network via <literal>podman
+network reload --all</literal>. If netavark version 1.9.0 or later is used for
+podman networking, then the
+<systemitem>netavark-firewalld-reload.service</systemitem> service can be used
+to solve this issue too. Enable and start the service as follows:
```
Comment (dmpop) [inline container-guide/concepts/containers-podman.xml:418]:
```suggestion
A temporary workaround is to reload the &podman; network using the <command>podman
network reload --all</command> command. Starting with version 1.9.0 of &netavark, you can use the
<systemitem>netavark-firewalld-reload.service</systemitem> service to solve this issue. Enable and start the service as follows:
```

**Review by dmpop — CHANGES_REQUESTED** [review]
A few language tweaks here and there.

**Comment by dcermak** [comment]
I've added a few additional sentences clarifying that this will **only** work if you are using netavark as the network subsystem. With CNI you're screwed.

File: container-guide/concepts/containers-podman.xml [hunk 4]
```diff
@@ -403,14 +403,33 @@
...
 </title>
 <para>
-Rootful containers may occasionally lose network connectivity due to another process deleting the firewall rules configured by &netavark;. Starting with version 1.9.0, &netavark; comes with the <systemitem>netavark-firewalld-reload.service</systemitem> service that solves the problem. Enable and start the service as follows:
+Rootful containers may occasionally become inaccessible from the outside. The
+issue is caused by firewalld reloading its permanent rules and discarding any
+temporary rules created by &podman;'s networking backend (either CNI or
+&netavark;).
+
+A temporary workaround is to reload the &podman; network using the
+<command>podman network reload --all</command> command. If you are using
```
Comment (danishprakash) [inline container-guide/concepts/containers-podman.xml:415]: The current sentence flows from "A temporary workaround.." so I feel it's better to distinguish it with the permanent fix now offered via the service. Wdyt?
```suggestion
<command>podman network reload --all</command> command.

A permanent solution exists if you are using...
```

File: container-guide/concepts/containers-podman.xml [hunk 5]
```diff
@@ -403,14 +403,33 @@
...
 </para>
 <screen>&prompt.root;systemctl enable netavark-firewalld-reload.service
 &prompt.root;systemctl restart netavark-firewalld-reload.service
 </screen>
+You can check which backend and version you are using by running
+<replaceable>podman info --format "{{.Host.NetworkBackend}}"</replaceable> and
+<replaceable>podman info --format
+"{{.Host.NetworkBackendInfo.Version}}"</replaceable>, respectively.
+
+We recommend to add permanent firewall rules for containers you want to be
```
Comment (danishprakash) [inline container-guide/concepts/containers-podman.xml:428]: Is this about users running containers with long-running jobs without frequently stopping and rerunning containers?

Comment (dcermak) [reply container-guide/concepts/containers-podman.xml:428]: Yes, for instance for a webpage. Such a container would be running all the time and it makes then sense to add a permanent firewall rule for it.

**Pull request merged**